cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java

```diff
@@ -628,45 +627,32 @@ private boolean processAddNodeLocally(Node node, AddNodeResponse response,
       }
 
       // check status of the new node
-//      AsyncClient client = (AsyncClient) connectNode(node);
-//      try {
-//        AsyncMethodCallback<String> result = new AsyncMethodCallback<String>() {
-//          String words;
-//          @Override
-//          public void onComplete(String s) {
-//            this.words = s;
-//          }
-//          @Override
-//          public void onError(Exception e) {
-//            System.out.println("some error happens");
-//          }
-//          public String getWords() {
-//            return words;
-//          }
-//        };
-//        System.out.println(result);
-//        client.echo("hello world", result);
-//      } catch (TException e) {
-//        e.printStackTrace();
-//      }
       CheckStatusRequest checkStatusRequest = new CheckStatusRequest();
       checkStatusRequest.setHashSalt(ClusterConstant.HASH_SALT);
       checkStatusRequest
           .setPartitionInterval(IoTDBDescriptor.getInstance().getConfig().getPartitionInterval());
       checkStatusRequest.setReplicationNumber(config.getReplicationNum());
-      CheckStatusHandler checkStatusHandler = new CheckStatusHandler();
+
+      AtomicReference<CheckStatusResponse> checkStatusResponseReference = new AtomicReference<>();
+      GenericHandler<CheckStatusResponse> checkStatusHandler = new GenericHandler<>(node,
+          checkStatusResponseReference);
       try {
-        sendStatusToNewNode(node, checkStatusRequest, checkStatusHandler);
-      } catch (TException exception) {
+        synchronized (checkStatusResponseReference) {
+          AsyncClient client = (AsyncClient) connectNode(node);
+          client.checkStatus(checkStatusRequest, checkStatusHandler);
+          checkStatusResponseReference.wait(connectionTimeoutInMS);
+        }
+      } catch (TException | InterruptedException exception) {
         logger.error("Failed to send current state to the new node {}", node, exception);
       }
-      if (!checkStatusHandler.getCheckStatusResponse().isPartitionalIntervalEquals()) {
+      CheckStatusResponse checkStatusResult = checkStatusResponseReference.get();
+      if (!checkStatusResult.isPartitionalIntervalEquals()) {
         logger.info("The partition interval of the new node {} conflicts.", node);
         return true;
-      } else if (!checkStatusHandler.getCheckStatusResponse().isHashSaltIntervalEquals()) {
+      } else if (!checkStatusResult.isHashSaltIntervalEquals()) {
         logger.info("The hash salt of the new node {} conflicts.", node);
         return true;
-      } else if (!checkStatusHandler.getCheckStatusResponse().isReplicationNumEquals()) {
+      } else if (!checkStatusResult.isReplicationNumEquals()) {
         logger.info("The replication number of the new node {} conflicts.", node);
         return true;
```

Comment:
If you decide to return true, then you must call either `resultHandler.onComplete()` or `resultHandler.onError()`, otherwise the requestor will wait forever. 
And be sure to let the requestor know why there is a failure.

Reply:
Thanks for your reminding. The onComplete() method will be called before returning results.